pkg/process/metadata/parser/java/yaml.go

```diff
@@ -0,0 +1,51 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package javaparser
+
+import (
+	"io"
+	"strings"
+
+	"gopkg.in/yaml.v3"
+)
+
+type flattener struct {
+	props map[string]string
+}
+
+func (f *flattener) visit(path []string, node *yaml.Node) {
+	for i := 0; i < len(node.Content); i += 2 { // should be safer
+		current := node.Content[i]
+		next := node.Content[i+1]
+		if current.Kind != yaml.ScalarNode || !(next.Kind == yaml.ScalarNode || next.Kind == yaml.MappingNode) {
+			continue
+		}
+		path = append(path, current.Value)
+
+		if next.Kind == yaml.MappingNode {
+			f.visit(path, next)
+		} else {
+			f.props[strings.Join(path, ".")] = next.Value // no need to decode... only support primitives not array
+		}
+		path = path[:len(path)-1]
+
+	}
+}
+
+// UnmarshalYAML custom yaml unmarshal
+func (f *flattener) UnmarshalYAML(value *yaml.Node) error {
```

Comment:
what's the performance impact of the method? I suspect there are a lot of allocations which can lead to high memory and cpu pressure

Reply:
I'm no expert for spring applications, but generally I'd stick with the simplification if the feature is not widely used

Reply:
it's generally speaking used so it's impossible to know in advance if a user will source `spring.application.name` from another property. I propose to keep this logic as it since it will address this case